netket/sampler/exact.py

```diff
@@ -83,6 +86,9 @@ def _reset(sampler, machine, parameters, state):
         )
         pdf = pdf / pdf.sum()
 
+        if config.netket_experimental_sharding:
+            pdf = gather(pdf)[: sampler.hilbert.n_states]
```

Comment:
can't this be handled by the argument `allgather=True`of `nn.to_array()`? it's already True by default and should guarantee this. I think you could copy-paste those lines in there.

Reply:
By the way, this way there is no need to add the code to handle sharding in the operators to sparse representation.
Don't trash it! But remove those changes from this PR.